tests/test_merge.py

```diff
@@ -484,3 +486,23 @@ def test_merge_allow_objects() -> None:
     cfg._set_flag("allow_objects", True)
     ret = OmegaConf.merge(cfg, {"foo": iv})
     assert ret == {"a": 10, "foo": iv}
+
+
+@pytest.mark.parametrize(  # type:ignore
+    "dst, other, expected",
+    [
+        (
+            OmegaConf.structured(InterpolationList),
+            OmegaConf.create({"list": [0.1]}),
+            {"list": [0.1]},
+        ),
+        (
+            OmegaConf.structured(InterpolationDict),
+            OmegaConf.create({"dict": {"a": 4}}),
+            {"dict": {"a": 4}},
+        ),
+    ],
+)
+def test_merge_with_interpolation(dst: Any, other: Any, expected: Any) -> None:
```

Comment:
Add a test for merging with itself.

Should be something like:
```
            OmegaConf.structured(InterpolationDict),
            OmegaConf.structured(InterpolationDict),
            {"dict": {"a": "${optimization.lr}"}},
```
I want to see that the interpolation is not resolved in the test.

Reply:
This fails, in this case in validate_merge

Reply:
<img width="895" alt="image" src="https://user-images.githubusercontent.com/30913090/98397516-c429fc80-205f-11eb-9d7d-94bac349d5e0.png">

Reply:
Okay, this needs to pass too. no reason for it to fail.